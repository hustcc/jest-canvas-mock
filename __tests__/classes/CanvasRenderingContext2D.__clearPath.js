let ctx;
beforeEach(() => {
  // get a new context each test
  ctx = document.createElement('canvas').getContext('2d');
});

afterEach(() => {
  const events = ctx.__getPath();
  expect(events).toMatchSnapshot();
});

describe('__clearEvents', () => {
  it('should clear the list of events', () => {
    ctx.arc(1, 2, 3, 4, 5);
    ctx.__clearPath();
  });

  it('should not prevent additional events from being collected', () => {
    ctx.arc(1, 2, 3, 4, 5);
    ctx.__clearPath();
    ctx.arc(1, 2, 3, 4, 5);
  });
});
