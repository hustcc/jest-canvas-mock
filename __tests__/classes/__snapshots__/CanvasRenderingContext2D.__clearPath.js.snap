// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__clearEvents should clear the list of events 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;

exports[`__clearEvents should not prevent additional events from being collected 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "anticlockwise": false,
      "endAngle": 5,
      "radius": 3,
      "startAngle": 4,
      "x": 1,
      "y": 2,
    },
    "type": "arc",
  },
]
`;
